src/Week1/HW01.hs

```diff
@@ -14,38 +13,80 @@ dropLastDigit n = n `div` 10
 
 -- Exercise 2 -----------------------------------------
 
+{- toRevDigits should convert positive Integers to a list of digits. (For
+0 or negative inputs, toRevDigits should return the empty list.) -}
+
+-- non tail recursive
 toRevDigits :: Integer -> [Integer]
-toRevDigits n = helper n []
+toRevDigits n 
+        | n <= 0 = [] 
+        | otherwise = lastDigit n : toRevDigits (dropLastDigit n)
+
+-- tail recursive version, assuming reverse is tail recursive
+toRevDigits' :: Integer -> [Integer]
+toRevDigits' n = helper n []
     where
-        helper x acc 
-            | x <= 0 = reverse acc 
+        helper :: Integer -> [Integer] -> [Integer]
+        helper x acc
+            | x <= 0 = reverse acc
+            | otherwise = helper (dropLastDigit x) (lastDigit x : acc)
+
+-- tail recursive non reversed digits
+toDigits :: Integer -> [Integer]
+toDigits n = helper n []
+    where
+        helper :: Integer -> [Integer] -> [Integer]
+        helper x acc
+            | x <= 0 = acc
             | otherwise = helper (dropLastDigit x) (lastDigit x : acc)
     
 
 -- Exercise 3 -----------------------------------------
 
 -- Double every second number in a list starting on the left.
+-- If only there was a mapi in haskell
+
 doubleEveryOther :: [Integer] -> [Integer]
 doubleEveryOther lst = helper 0 lst [] 
     where 
+        helper :: Integer -> [Integer] -> [Integer] -> [Integer]
         helper ctr trav acc
-            | trav == [] = reverse acc
+            | null trav = reverse acc
```

Comment:
Using pattern matching would be better than using guards + head and tail functions.
Also [odd](https://hackage.haskell.org/package/base-4.15.0.0/docs/Prelude.html#v:odd) ctr is preferrable to `mod ctr 2 == 1`.

```haskell
helper ctr [] acc = reverse acc
helper ctr (x:xs) acc
    | odd ctr = ...
```

Alternatives to expand your thinking: you can avoid explicit recursion and custom counter parameter by combining [zipWith](https://hackage.haskell.org/package/base-4.15.0.0/docs/Prelude.html#v:zipWith) and [cycle](https://hackage.haskell.org/package/base-4.15.0.0/docs/Prelude.html#v:cycle) in an intelligent way.

<details>
  <summary>spoiler solution</summary>
  <pre>
doubleEveryOther = zipWith (*) (cycle [1,2])
  </pre>
</details>

Reply:
Added your suggestions in with alternative functions for exercise 3. The usage of cycle is pretty smart too.